docker-compose.yml

```diff
@@ -0,0 +1,32 @@
+
+services:
+  go_app:
+    build:
+      context: ./
+    volumes:
+      - ./templates:/templates # Mount templates for the Go app
+    networks:
+      - app_network
+    container_name: go_app
+    expose:
+      - "8080" # Expose internal port for Nginx to connect
+    environment:
+      - PORT=8080 # Example: Set your app's listening port if required
+
+  nginx_proxy:
+    build:
+      context: ./nginx
+    ports:
+      - "80:80" # Map Nginx's port 80 to host port 80
+    networks:
+      - app_network
+    depends_on:
+      - go_app
+    container_name: nginx_proxy
+    volumes:
+      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+    entrypoint: [ "tail", "-f", "/dev/null" ] # Keep container running for debugging
```

Comment:
This custom `entrypoint` is the problem.

When you do this, you overwrite the original one that calls nginx service.

Removing it, you'll notice an error like this:

![image](https://github.com/user-attachments/assets/b223938a-b747-4857-8041-e96ff0c4bf0b)